src/peft/peft_model.py

```diff
@@ -531,56 +530,13 @@ def load_adapter(self, model_id: str, adapter_name: str, is_trainable: bool = Fa
                 model_id,
                 **hf_hub_download_kwargs,
             )
-            if isinstance(peft_config, PromptLearningConfig) and is_trainable:
+            if peft_config.is_prompt_learning and is_trainable:
                 raise ValueError("Cannot set a prompt learning adapter to trainable when loading pretrained adapter.")
             else:
                 peft_config.inference_mode = not is_trainable
             self.add_adapter(adapter_name, peft_config)
 
-        # load weights if any
-        path = (
-            os.path.join(model_id, hf_hub_download_kwargs["subfolder"])
-            if hf_hub_download_kwargs.get("subfolder", None) is not None
-            else model_id
-        )
-
-        if os.path.exists(os.path.join(path, SAFETENSORS_WEIGHTS_NAME)):
-            filename = os.path.join(path, SAFETENSORS_WEIGHTS_NAME)
-            use_safetensors = True
-        elif os.path.exists(os.path.join(path, WEIGHTS_NAME)):
-            filename = os.path.join(path, WEIGHTS_NAME)
-            use_safetensors = False
-        else:
-            has_remote_safetensors_file = hub_file_exists(
-                model_id,
-                SAFETENSORS_WEIGHTS_NAME,
-                revision=hf_hub_download_kwargs.get("revision", None),
-                repo_type=hf_hub_download_kwargs.get("repo_type", None),
-            )
-            use_safetensors = has_remote_safetensors_file
-
-            if has_remote_safetensors_file:
-                # Priority 1: load safetensors weights
-                filename = hf_hub_download(
-                    model_id,
-                    SAFETENSORS_WEIGHTS_NAME,
-                    **hf_hub_download_kwargs,
-                )
-            else:
-                try:
-                    filename = hf_hub_download(model_id, WEIGHTS_NAME, **hf_hub_download_kwargs)
-                except EntryNotFoundError:
-                    raise ValueError(
-                        f"Can't find weights for {model_id} in {model_id} or in the Hugging Face Hub. "
-                        f"Please check that the file {WEIGHTS_NAME} or {SAFETENSORS_WEIGHTS_NAME} is present at {model_id}."
-                    )
-
-        if use_safetensors:
-            adapters_weights = safe_load_file(filename, device="cuda" if torch.cuda.is_available() else "cpu")
-        else:
-            adapters_weights = torch.load(
-                filename, map_location=torch.device("cuda" if torch.cuda.is_available() else "cpu")
-            )
+        adapters_weights = self._get_peft_state_dict(model_id, **hf_hub_download_kwargs)
```

Comment:
as `_get_peft_state_dict` is loading weights, the naming here is confusing. It should be `_load_peft_weights` or something along those lines

Reply:
Agreed, I have moved the method to `save_and_load` utils and renamed it

Reply:
Also made the method public so that it can be used for the transformers integration and more generally by users